### ISSUE TITLE
File Explorer opening on launch, game not opening

### ISSUE THREAD

**Issue by kbdevs** [Heroic-Games-Launcher/HeroicGamesLauncher#2832]
### Describe the bug

Hey everyone, I'm trying to launch a game through heroic and its just launching a file explorer of my prefix folder, any ideas what could I do to fix this?

(Some games have worked on windows version steam)

### Add logs

```text
(18:38:37) INFO:    [Backend]:          Adding shortcuts for PC Building Simulator

(18:38:37) INFO:    [DownloadManager]:  Finished Installation of ab277c0995e945d2b2c50c46883627f1

(18:38:37) INFO:    [DownloadManager]:  ab277c0995e945d2b2c50c46883627f1 added to download manager finished.

(18:38:37) INFO:    [DownloadManager]:  ab277c0995e945d2b2c50c46883627f1 removed from download manager.

(18:38:37) INFO:    [Frontend]:         Refreshing legendary Library

(18:38:37) INFO:    [Legendary]:        Refreshing library...

(18:38:37) INFO:    [Legendary]:        Refreshing Epic Games...

(18:38:37) INFO:    [Legendary]:        Game list updated, got 7 games & DLCs

(18:38:37) INFO:    [Backend]:          Stopping Power Saver Blocker

(18:38:37) DEBUG:   [Legendary]:        Using cached install info

(18:38:38) INFO:    [Legendary]:        Checking if EOS Overlay is enabled: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary eos-overlay info

(18:38:38) DEBUG:   [Legendary]:        Using cached install info

(18:38:38) DEBUG:   [Legendary]:        Using cached install info

(18:38:38) DEBUG:   [Legendary]:        Using cached install info

(18:38:38) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary list --third-party

(18:39:26) INFO:    [Backend]:          Launching PC Building Simulator (ab277c0995e945d2b2c50c46883627f1)

(18:39:26) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary --version

(18:39:26) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7

(18:39:26) DEBUG:   [Legendary]:        Using cached install info

(18:39:26) INFO:    [Backend]:          Stopping Power Saver Blocker

(18:39:26) DEBUG:   [Legendary]:        Using cached install info

(18:39:26) DEBUG:   [Legendary]:        Using cached install info

(18:39:26) DEBUG:   [Legendary]:        Using cached install info

(18:39:26) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7

(18:39:26) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7

(18:39:26) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7

(18:39:26) DEBUG:   [Backend]:          Running Wine command: wineboot --init

**(18:39:31) WARNING: [Backend]:          Could not determine System Info**

(18:39:34) INFO:    [Gog]:              Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl --auth-config-path "/Users/kb/Library/Application Support/heroic/gog_store/auth.json" --version

(18:39:36) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary launch ab277c0995e945d2b2c50c46883627f1 --json --offline

(18:39:50) INFO:    [Legendary]:        Launching PC Building Simulator: LD_PRELOAD= WINEESYNC=1 /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary launch ab277c0995e945d2b2c50c46883627f1 --language en --wrapper "/usr/local/Cellar/game-porting-toolkit/1.0/bin/wine64 /Users/kb/my-game-prefix" --no-wine

(18:40:29) INFO:    [Backend]:          Stopping Display Power Saver Blocker

(18:40:29) INFO:    [Frontend]:         Refreshing legendary Library

(18:40:29) INFO:    [Legendary]:        Refreshing library...

(18:40:29) INFO:    [Legendary]:        Refreshing Epic Games...

(18:40:29) INFO:    [Legendary]:        Game list updated, got 7 games & DLCs

(18:40:29) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary list --third-party
```

### Steps to reproduce

1. Hit play on the game
2. Wait
3. Show File Explorer
4. Nothing Launches

### Expected behavior

The game should launch

### Heroic Version

main branch from GitHub

### System Information

- OS [e. g. "Ubuntu"]: MacOS 14.0 Sonoma

### Additional information

Trying to use Game porting toolkit

**Comment by pakkographic** [comment]
same issue here

**Comment by pakkographic** [comment]
<img width="744" alt="screenshot" src="https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/41129374/bba66f85-b902-447b-9b80-7761c1617bda">

**Comment by blackxfiied** [comment]
can confirm this is a problem that happens to me as well, here's the command that gets run by heroic:

`
LD_PRELOAD= /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary launch 48619f5656da446abccb727b78cda644 --language en --wrapper "/usr/local/Cellar/game-porting-toolkit/1.0.2/bin/wine64 /Users/blackxfiied/Library/Containers/com.isaacmarovitz.Whisky/Bottles/6FAC3D2F-340D-453A-95AF-9D5F5B85D66F" --no-wine
`

dont mention whisky i've tried not using whisky

**Comment by blackxfiied** [comment]
Found the problem, it's becuase of the --wrapper argument. instead of referencing the "gameportingtoolkit" script in /usr/local/bin, heroic references gameportingtoolkit's wine64 (wine 7.7)

here's what heroic tries to execute according to logs (game launched is pikuniku.)
`LD_PRELOAD= /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary launch 41869934302e4b8cafac2d3c0e7c293d --language en --wrapper "/usr/local/Cellar/game-porting-toolkit/1.0.2/bin/wine64 /Users/blackxfiied/my-game-prefix" --no-wine`

this wont work becuase its trying to reference a folder to wine, which will just open the folder in file explorer

what does work is
`LD_PRELOAD= /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary launch 48619f5656da446abccb727b78cda644 --language en --wrapper "/usr/local/bin/gameportingtoolkit /Users/blackxfiied/my-game-prefix" --no-wine`

becuase the game poprting toolkit script executes it instead of feeding it directly to wine

**Comment by arielj** [comment]
can anyone in this thread confirm if this got fixed in 2.9.1?

**Comment by arielj** [comment]
Closing this since I understand https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/2880 fixes this issue
